packages/core/src/amazonqFeatureDev/client/featureDev.ts

```diff
@@ -168,6 +174,12 @@ export class FeatureDevClient {
                         e.$metadata.requestId ?? 'unknown'
                     }`
                 )
+                if (
+                    e.name === 'ThrottlingException' &&
+                    e.message.includes('limit for number of iterations on an implementation plan')
```

Comment:
We had discussed returning ServiceQuotaExceededException from RTS to toolkits before re:Invent and they were not keen to return it. They insisted on returning ThrottlingException with [a specific error message](https://code.amazon.com/packages/AWSVectorConsolasRuntimeService/blobs/7ae73dbe13dde0e1974f5c76e8ac8a8c38cbe064/--/src/com/amazon/aws/vector/consolas/runtimeservice/utils/ExceptionConstants.java#L25). 

I agree that string matching isn't a good idea. Do you think that we should reconsider it with RTS folks maybe come up with a better approach?

Reply:
Yes I think we should revisit this. It's not clear if their preference for using `ThrottlingException` was because of time constraints or a different reason.